### PR TITLE
MWPW-144811: custom params in co links

### PR DESCRIPTION
**Pull request by yesil**
Brings support for custom checkout link parameters to merch links.

They are stored on the `data-extra-options` attribute on the checkout links and passed onto the checkout-url-builder.

Includes: 
 - https://git.corp.adobe.com/wcms/tacocat.js/pull/565
 - https://git.corp.adobe.com/wcms/tacocat.js/pull/566

Resolves: [MWPW-144811](https://jira.corp.adobe.com/browse/MWPW-144811)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ilyas/checkout-links?martech=off&georouting=off
- After: https://mwpw-144811--milo--yesil.hlx.page/drafts/ilyas/checkout-links?martech=off&georouting=off

**More test URLs**
- https://mwpw-144811--milo--yesil.hlx.page/drafts/mili/checkout-links?martech=off&georouting=off

**Screenshots:**
image

image

### CODE REVIEW

File: test/blocks/merch/merch.test.js [hunk 1]
```diff
@@ -603,4 +600,26 @@ describe('Merch Block', () => {
       document.querySelector('.modal-curtain').click();
     });
   });
+
+  describe.only('checkout link with optional params', async () => {
```
Comment (mokimo) [inline test/blocks/merch/merch.test.js:604]: U accidently kept the `.only`

**Review by mokimo — APPROVED** [review]
Remove the `.only` from the merch test file, otherwise it looks good 👍

File: test/blocks/merch/merch.test.js [hunk 2]
```diff
@@ -603,4 +600,26 @@ describe('Merch Block', () => {
...
+    const CHECKOUT_ALLOWED_KEYS_VALUES = { mal: 2 };
+    CHECKOUT_ALLOWED_KEYS.forEach((key, index) => {
+    // ['usid'].forEach((key, index) => {
```
Comment (Axelcureno) [inline test/blocks/merch/merch.test.js:607]: is this needed?

**Review by Axelcureno — APPROVED** [review]

**Comment by skumar09** [comment]
@yesil:  On your PR Branch : `mwpw-144811--milo--yesil.hlx.live ` Georouting tests are failing ( i.e., Georouting modal is not showing), same is working on milo-main. Can you please review
 https://mwpw-144811--milo--yesil.hlx.live/de/drafts/nala/features/georouting/georouting  ( Georouting modal is not displaying)
https://main--milo--adobecom.hlx.live/de/drafts/nala/features/georouting/georouting ( Georouting modal is displaying) (edited)

File: test/blocks/merch/merch.test.js (hunk 1 above)
Comment (yesil) [reply test/blocks/merch/merch.test.js]: thanks, removed.

File: test/blocks/merch/merch.test.js (hunk 2 above)
Comment (yesil) [reply test/blocks/merch/merch.test.js]: removed, thx.

**Comment by Roycethan** [comment]
@afmicka Plz feel free to check this out here is the list of params : https://github.com/yesil/milo/blob/MWPW-144811/libs/blocks/merch/merch.js#L39-L74

**Comment by Roycethan** [comment]
@yesil Updated here https://jira.corp.adobe.com/browse/MWPW-136885?focusedCommentId=41043352&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-41043352
Plz review thanks

**Comment by afmicka** [comment]
> @yesil: On your PR Branch: `mwpw-144811--milo--yesil.hlx.live ` Georouting tests are failing ( i.e., Georouting modal is not showing), same is working on milo-main. Can you please review https://mwpw-144811--milo--yesil.hlx.live/de/drafts/nala/features/georouting/georouting ( Georouting modal is not displaying) https://main--milo--adobecom.hlx.live/de/drafts/nala/features/georouting/georouting ( Georouting modal is displaying) (edited)
> 
> **Info:** Other failures related to browser console error: 'Attestation check of topics on 'URL' `, I am updating the nala tests to ignore them. Please let me know if it is okay to ignore them.

@nateekar georouting tests fail on every PR raised from the fork (like this one, milo--yesil) due to the CORS. They work only from the upstream branches (milo--adobecom).

File: libs/blocks/merch/merch.js [hunk 3]
```diff
@@ -34,6 +34,46 @@ export const CC_SINGLE_APPS = [
...
+  'appctxid',
+  'context.guid',
+  // 'countrySpecific', //TODO not allowed in checkout url builder.
+  'csm',
+  'ctxRtUrl',
+  'DCWATC',
+  'dp', // Enable digital payments for iframe context
+  'fr', // represents the commerce app redirecting to UC
+  'gsp',
+  'jit',
```
Comment (afmicka) [inline libs/blocks/merch/merch.js:50]: could this be a typo? There is 'ijt' parameter in the wiki but not 'jit'.

**Review by afmicka — COMMENTED** [review]
the added parameters work. Left a [comment](https://jira.corp.adobe.com/browse/MWPW-144811#:~:text=added%20a-,comment,-%2D%205%20minutes) in jira to verify if parameters that are in the wiki but not supported with the PR are needed any time soon. Parameters in question are - ijt, pa, ms, ot, cs, spint. cc: @Roycethan

**Comment by afmicka** [comment]
@yesil the only parameter i am not managing to get is 'cs' (customer segment for segmentation link, in [wiki](https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=businessservices&title=UCv3+Link+Creation+Guide#:~:text=schools%20and%20universities%22-,cs,-String)). This might also be a typo (but not sure) in the [list](https://github.com/yesil/milo/blob/179b5ad0a310480ae28faf0ad7f88524b0a38466/libs/blocks/merch/merch.js#L45) where you have 'csm' (not sure what that one is).

The rest looks good to me.

@Roycethan please check on your end too.

**Pull request merged**